neuro_evolution_ctrnn/tools/env_handler.py

```diff
@@ -78,45 +74,85 @@ def make_env(self, env_id: str):
-        self.screen_size = 32
+class ProcEnvHandler(gym.Env):
+    """
+    This Wrapper scales to observation to values between 0 and 1.
+    Additionally it implements a seed method because for reasons unknown it not implemented upstream
+    """
+
+    def __init__(self, env_id, render):
+        # todo: maybe add env specific configuration, but only after issue #20 has been implemented
+        self.env_id = env_id
+        self.render_mode = None
+        if render:
+            self.render_mode = 'rgb_array'
+        super(ProcEnvHandler, self).__init__()
+        self._env = self._make_inner_env(start_level=0)
+        self.spec = copy.deepcopy(self._env.spec)  # deep copy to avoid references to inner gym
+        self.action_space = self._env.action_space  # use reference, so action_space.seed() works as expected
         self.obs_dtype = np.float16
+        self.input_high = 255
+        self.current_level = 0
+        assert self.input_high == self._env.observation_space.high.min(), "unexpected bounds for input space"
+        assert self.input_high == self._env.observation_space.high.max(), "unexpected bounds for input space"
+        assert 0 == self._env.observation_space.low.min(), "unexpected bounds for input space"
+        assert 0 == self._env.observation_space.low.max(), "unexpected bounds for input space"
         self.observation_space = Box(low=0, high=1,
-                                     shape=(self.screen_size, self.screen_size, 3),
+                                     shape=self._env.observation_space.shape,
                                      dtype=self.obs_dtype)
 
+    def _make_inner_env(self, start_level):
+        self.current_level = start_level
+        env = gym.make(self.env_id,
+                       distribution_mode="memory",
+                       use_monochrome_assets=False,
+                       restrict_themes=True,
+                       use_backgrounds=False,
+                       num_levels=1,
+                       start_level=self.current_level,
+                       render_mode=self.render_mode
+                       )
+        return env
+
     def _transform_ob(self, ob):
-        ob = cv2.resize(ob, (self.screen_size, self.screen_size), interpolation=cv2.INTER_AREA)
         return np.asarray(ob, dtype=self.obs_dtype) / 255.0
 
+    def render(self, mode='human', **kwargs):
+        frame = self._env.render(mode=self.render_mode, **kwargs)
+        cv2.imshow("ProcGen Agent", frame)
+        cv2.waitKey(1)
+
     def step(self, action):
-        ob, rew, done, info = super(ProcEnvWrapper, self).step(action)
+        ob, rew, done, info = self._env.step(action)
         return self._transform_ob(ob), rew, done, info
 
     def reset(self):
-        return self._transform_ob(super(ProcEnvWrapper, self).reset())
+        del self._env
+        self._env = self._make_inner_env(start_level=self.current_level + 1)
+        return self._transform_ob(self._env.reset())
```

Comment:
Ist das `self.current_level` wichtig für die Evaluation/Ergebnisse/etc.? Falls ja wäre es wichtig zu wissen dass bei multiprocessing im Moment immer eine neue Environment erstellt wird, pro Aufruf von `eval_fitness()`. Das bedeutet, da erhöht sich das Level potentiell immer nur dann wenn `config.episode_runner.number_fitness_runs > 1`, da dann die Env mehrmals benutzt wird.

Reply:
> Das bedeutet, da erhöht sich das Level potentiell immer nur dann wenn config.episode_runner.number_fitness_runs > 1, da dann die Env mehrmals benutzt wird.

ja, genau. 

proc gen ist normalerweise so gemacht, dass immer ein zufälliges level ausgewählt wird zwischen `startlevel` und `startlevel+num_level`. Und man hat (glaube ich) keine möglichkeit eine reihenfolge der level festzulegen. Um eine definierte reihenfolge der level festzulegen, habe ich `num_level=1` gesetzt und hier in der reset methode wird dann das nächste level ausgewählt. 

Ich denke wenn wir anfangen memoryexperimente zu machen, wo jedes level 2 mal durchlaufen wird, muss man hier vllt.  was ändern. Aber darüber mache ich mir erst gedanken ,wenn es soweit ist. ;-)